FWCore/Framework/src/RunProcessingStatus.cc

```diff
@@ -0,0 +1,44 @@
+// -*- C++ -*-
+//
+// Package:     FWCore/Framework
+// Class  :     RunProcessingStatus
+//
+// Original Author:  W. David Dagenhart
+//         Created:  7 October 2021
+
+#include "RunProcessingStatus.h"
+#include "FWCore/Framework/interface/RunPrincipal.h"
+
+namespace edm {
+  RunProcessingStatus::RunProcessingStatus(unsigned int iNStreams, WaitingTaskHolder const& holder)
+      : holderOfTaskInProcessRuns_(holder), nStreamsStillProcessingRun_(iNStreams) {}
+
+  void RunProcessingStatus::resetBeginResources() {
+    endIOVWaitingTasks_.doneWaiting(std::exception_ptr{});
+    for (auto& iter : eventSetupImpls_) {
+      iter.reset();
+    }
+  }
+
+  void RunProcessingStatus::resetEndResources() {
+    endIOVWaitingTasksEndRun_.doneWaiting(std::exception_ptr{});
+    for (auto& iter : eventSetupImplsEndRun_) {
+      iter.reset();
+    }
+  }
+
+  void RunProcessingStatus::setEndTime() {
+    if (2 != endTimeSetStatus_) {
```

Comment:
Instead of using a magic number, how about a named constexpr variable?

Reply:
Done. Thanks.

I fixed the function in LuminosityBlockProcessingStatus also which I had lazily just copied.